utils/githooks/commit-msg.d/10-watermark.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+"""Add watermark to commit message"""
+import os
+import sys
+
+
+def check_if_run(name):
+    """Check existence of a file, removes file"""
+    if not os.path.exists(f".{name}"):
+        print(f"Required githook {name} is not installed. See utils/githooks/README.md")
+        return False
+    os.remove(f".{name}")
+    return True
+
+
+def find_hooks():
+    """Find the required git hooks"""
+    hooks = []
+    for fname in os.listdir('utils/githooks'):
+        if fname == "commit-msg.d":
+            continue
+        if not fname.endswith('.d'):
+            continue
+        if not os.path.isdir(os.path.join("utils/githooks")):
+            continue
+        hooks.append(fname[:-2])
+    return hooks
+
+
+def run_check():
+    """Run the checks for the required commit hooks"""
+    empty_commit = True
+    msg = []
+    with open(sys.argv[1], "r") as commit_msg:
+        for line in commit_msg.readlines():
+            if line.startswith("Required-githooks: true"):
+                sys.exit(0)
+            elif line.startswith("Signed-off-by"):
+                continue
+            elif line.strip() == "":
+                continue
+            elif line.startswith("#"):
+                break
+            empty_commit = False
+    if empty_commit:
+        sys.exit(0)
+
+    hooks = find_hooks()
+    missing = False
+    for hook in hooks:
+        if not check_if_run(hook):
+            missing = True
+    if missing:
+        sys.exit(0)
```

Comment:
Optional - For a compact solution...
```suggestion
    if not all(map(check_if_run, hooks)):
        sys.exit(0)
```

Reply:
Or, to forego lazy evaluation and ALWAYS run check_if_run:
```
    if not all(list(map(check_if_run, hooks))):
        sys.exit(0)
```

Reply:
I'll defer these things to a later solution. I definitely do not want it to be lazy.  I would have broken the loop in such case.